### PR TITLE
More Python 3 porting 

### PR DESCRIPTION
**Pull request by hawkowl**
This also improves the tox.ini by making it get the dependencies from the current setup.py, not the one published on PyPI.

### CODE REVIEW

**Comment by oberstet** [comment]
why a different ordered set? I just removed it;)

**Comment by hawkowl** [comment]
I stopped using the other one when I realised why it was failing :)

**Comment by oberstet** [comment]
well, we don't need any external dep for that .. I am wondering why this PR isn't in conflict, since I just yesterday removed the other ordered_set. anyway, I'm not gonna merge it like this

**Comment by hawkowl** [comment]
I merged it forward, to get rid of the dependency :) -- the diff shouldn't have it there.

File: crossbar/router/observation.py [hunk 1]
```diff
@@ -43,8 +43,8 @@ def is_protected_uri(uri):
 
 class OrderedSet(set):
 
-    def __init__(self, iter=None):
-        super(set, self).__init__(iter)
+    def __init__(self):
```
Comment (oberstet) [inline crossbar/router/observation.py]: why?

**Comment by oberstet** [comment]
changes to tox.ini: are you sure these changes work?

File: crossbar/router/observation.py (hunk 1 above)
Comment (hawkowl) [inline crossbar/router/observation.py]: set() on Python 3 works differently, and we don't instantiate it with any arguments in the codebase (it also doesn't put iter in `self._list`).

**Comment by hawkowl** [comment]
I ran them all locally, and Travis is giving all-green. `skip_install` under flake8 requires a recent tox (1.9, current is 2.0.1), but it makes it run massively quicker (11s Travis job for flake8, master takes 54sec).

**Pull request merged**